src/main/java/org/opensearch/timeseries/feature/SearchFeatureDao.java

```diff
@@ -844,15 +891,221 @@ public List<Long> parseColdStartSampleTimestamp(SearchResponse response, boolean
+     * slide by one bucket at a time, so they *overlap* just like the classic time-series
+     * shingle used by Random Cut Forest.
+     *
+     * Why “just count non-empty buckets” is insufficient?
+     *
+     * <pre>
+     *   • Documents arrive every 70 min   →  0, 70, 140, 210, 280 …
+     *   • Query buckets are 58 min wide   →  0–58, 58–116, 116–174 …
+     *
+     *   0        58        116        174        232        290        348        406
+     *   │────────│─────────│─────────│─────────│─────────│─────────│─────────│────────
+     *   bucket 0   bucket 1   bucket 2   bucket 3   bucket 4   bucket 5   bucket 6 …
+     *                  ▲70          ▲140         ▲210         ▲280                 ▲350
+     * </pre>
+     *
+     * <ul>
+     *   <li>The first point (70 min) lands 12 min inside <em>bucket&nbsp;1</em>; bucket 0 is empty.</li>
+     *   <li>Every new point is 70 min after the previous one, i.e. 12 min deeper into the grid
+     *       (70 − 58 = 12).  Offsets therefore walk 0 → 12 → 24 → 36 → 48 → 60 min.  When the
+     *       offset reaches 60 min it “overflows” the 58-min bucket and the point jumps into
+     *       the next bucket, leaving a gap behind.</li>
+     *   <li>The result is a perfectly regular pattern: five buckets full, one empty, repeating.</li>
+     * </ul>
+     *
+     * Merely summing all non-empty buckets would say “34 out of 40 buckets contain data”,
+     * suggesting good coverage, yet <em>any</em> shingle that spans six or more buckets
+     * is broken.  The correct metric is therefore:
+     *
+     * <blockquote>“How many sliding windows of length {@code shingleSize}
+     *  are <strong>fully</strong> populated?”</blockquote>
+     *
+     * Examples with the above pattern:
+     * <ul>
+     *   <li>{@code shingleSize = 4} → one valid shingle (buckets 1-4).</li>
+     *   <li>{@code shingleSize = 3} → two valid shingles (1-3 and 2-4).</li>
+     *   <li>{@code shingleSize = 6} → zero valid shingles (every 6-bucket window crosses a gap).</li>
+     * </ul>
+     *
+     * Algorithm
+     * <ol>
+     *   <li>Extract and time-sort the buckets from the aggregation.</li>
+     *   <li>Turn the list into a boolean array where {@code true == doc_count > threshold}.</li>
+     *   <li>Slide a window of length {@code shingleSize}; keep a running count of how many
+     *       {@code true}s are inside the window.  Each time the window is “full size” and
+     *       the running count equals {@code shingleSize}, increment the shingle counter.</li>
+     * </ol>
+     *
+     * The implementation is O(<i>N</i>) time with O(<i>N</i>) temporary booleans
+     * (easily reducible to O(1) if memory is tight).
+     *
+     * @param response            the {@link SearchResponse} returned by OpenSearch
+     * @param config              configuration
+     * @param includesEmptyBucket if {@code true}, a bucket with {@code doc_count == 0}
```

Comment:
I think because we changed to mindoccount to 0 in other part of code this is always true? Additionally we don't seem to actually utilize this variable anywhere in method

Reply:
yes, removed